### ISSUE TITLE
Build button fails to build, but running make manually works

### ISSUE THREAD

**Issue by cdwilson** [microsoft/vscode-makefile-tools#249]
I'm getting the following error when trying to run make via the Build button:

```bash
make: *** "/Users/chris/Projects/micropython/ports/stm32/": No such file or directory.  Stop.
The terminal process "bash '-c', ''make' 'all' 'BOARD=FCD_MODST_MOUSE_V1' 'V=1' '-C' '"/Users/chris/Projects/micropython/ports/stm32/"''" failed to launch (exit code: 2).
```

![image](https://user-images.githubusercontent.com/14631/145074085-3abe9efc-55aa-41f8-a447-323db2d877cf.png)

Running the same command via the Terminal works fine:

![image](https://user-images.githubusercontent.com/14631/145074366-c17563a1-cc1e-44d7-8f07-858e2db4187c.png)

I think this might be some kind of escaping issue.  I tried a couple variations:

This works:
```
$ make all BOARD=FCD_MODST_MOUSE_V1 V=1 -C "/Users/chris/Projects/micropython/ports/stm32/"
python3 ../../py/makeversionhdr.py build-FCD_MODST_MOUSE_V1/genhdr/mpversion.h
python3 ../../tools/makemanifest.py -o build-FCD_MODST_MOUSE_V1/frozen_content.c -v "MPY_DIR=../.." -v "MPY_LIB_DIR=../../../micropython-lib" -v "PORT_DIR=/Users/chris/Projects/micropython/ports/stm32" -v "BOARD_DIR=boards/FCD_MODST_MOUSE_V1" -b "build-FCD_MODST_MOUSE_V1" -f"-march=armv7m" --mpy-tool-flags="" boards/manifest.py
```

This doesn't work:
```
$ bash -c make all BOARD=FCD_MODST_MOUSE_V1 V=1 -C "/Users/chris/Projects/micropython/ports/stm32/"
make: *** No targets specified and no makefile found.  Stop.
```

This works:
```
$ bash -c 'make all BOARD=FCD_MODST_MOUSE_V1 V=1 -C "/Users/chris/Projects/micropython/ports/stm32/"'
python3 ../../py/makeversionhdr.py build-FCD_MODST_MOUSE_V1/genhdr/mpversion.h
python3 ../../tools/makemanifest.py -o build-FCD_MODST_MOUSE_V1/frozen_content.c -v "MPY_DIR=../.." -v "MPY_LIB_DIR=../../../micropython-lib" -v "PORT_DIR=/Users/chris/Projects/micropython/ports/stm32" -v "BOARD_DIR=boards/FCD_MODST_MOUSE_V1" -b "build-FCD_MODST_MOUSE_V1" -f"-march=armv7m" --mpy-tool-flags="" boards/manifest.py
```

**Comment by threshold68** [comment]
Can confirm this bug for version 0.3. Reverting back to version 0.22 solves this issue. As @cdwilson I am also on macos albeit with zsh and with the makefile located in a different directory, i.e. I am using `"makefile.makeDirectory"` and "`makefile.makefilePath"` in `settings.json`. Seems that some changes introduced in version 0.3 broke escaping.

**Comment by andreeis** [comment]
@threshold68 and @cdwilson yes this is a regression and we are working on an urgent fix. We will probably ship a 0.3.1 with a few other bug fixes and as will share with you even sooner a private vsix that you can manually install and get unblocked until the next update of 0.3.1 on the Microsoft Market Place.

**Comment by andreeis** [comment]
@threshold68 and @cdwilson , please manually install the vsix from [this link](https://github.com/microsoft/vscode-makefile-tools/actions/runs/1556708172) and let me know how everything works. It fixes the quoting for when building the makefile.

**Comment by scorpadorp** [comment]
> @threshold68 and @cdwilson , please manually install the vsix from [this link](https://github.com/microsoft/vscode-makefile-tools/actions/runs/1556708172) and let me know how everything works. It fixes the quoting for when building the makefile.

Can confirm this works.

Thanks! I waisted several days wondering what was wrong.

**Comment by andreeis** [comment]
Closing this since we just released a fix with Makefile Tools 0.3.1 today.

**Comment by cdwilson** [comment]
@andreeis just tested the released 0.3.1 and it fixes the build issues for me.  Thanks for the quick fix on this!

**Comment by andreeis** [comment]
@cdwilson, I am glad to hear that. If you would like, you can also give us a review on [Microsoft Market Place](https://marketplace.visualstudio.com/items?itemName=ms-vscode.makefile-tools&ssr=false#review-details).
